staging/src/k8s.io/client-go/tools/pager/pager.go

```diff
@@ -245,3 +289,16 @@ func (p *ListPager) eachListChunk(ctx context.Context, options metav1.ListOption
 		options.Continue = m.GetContinue()
 	}
 }
+
+func isTransientError(err error) bool {
```

Comment:
Can you unit test this with all the errors it matches?

Reply:
done.

Reply:
I think our client  also retries on connection errors.